casanovo/denovo/model.py

```diff
@@ -1096,24 +1098,23 @@ def calc_match_score(
         :, :-1
     ]  # Remove trailing tokens from label, remove -1 to keep stop token
     all_scores = []
+    per_aa_scores = []
 
     for all_aa_pred, truth_indicies in zip(
         batch_all_aa_scores, truth_aa_indicies
     ):
         assert len(all_aa_pred) == len(
             truth_indicies
         )  # Ensure that length of score list and indexes to pull from are the same length
-        aa_score_sum = 0
+        aa_scores = []
         for scores, true_index in zip(all_aa_pred, truth_indicies):
```

Comment:
Nit: Would probably be cleaner to get per_aa_scores all at once using a pytorch function call to get all the needed indexes (eg. using torch.gather) rather than the nested for loop. Could bring this method down to a couple of lines.